src/sentry/api/endpoints/api_tokens.py

```diff
@@ -35,23 +37,36 @@ class ApiTokensEndpoint(Endpoint):
     authentication_classes = (SessionNoAuthTokenAuthentication,)
     permission_classes = (IsAuthenticated,)
 
-    @method_decorator(never_cache)
-    def get(self, request: Request) -> Response:
+    @classmethod
+    def _get_appropriate_user_id(cls, request: Request) -> str:
+        """
+        Gets the user id to use for the request, based on what the current state of the request is.
+        If the request is made by a superuser, then they are allowed to act on behalf of other user's data.
+        Therefore, when GET or DELETE endpoints are invoked by the superuser, we may utilize a provided user_id.
+
+        The user_id to use comes from the GET or BODY parameter based on the request type.
+        For GET endpoints, the GET dict is used.
+        For all others, the DATA dict is used.
+        """
+        # Get the user id for the user that made the current request as a baseline default
         user_id = request.user.id
         if is_active_superuser(request):
-            user_id = request.GET.get("userId", user_id)
+            datastore = request.GET if request.GET else request.data
```

Comment:
🤔 it is just me or is it a bit weird we sometimes send data as query params and other times as request data?

Reply:
Ohh, I think this is just misleading! I captured this in the comment:
```
The user_id to use comes from the GET or BODY parameter based on the request type.
        For GET endpoints, the GET dict is used.
        For all others, the DATA dict is used.
```
Does that help?

Reply:
i think so, but i'm more wondering why we don't always send data in `request.data`. maybe this is a convention i'm learning

Reply:
Django Rest framework reference to help give more context: https://www.django-rest-framework.org/api-guide/requests/

Reply:
It's actually saying we shouldn't use `GET` haha and instead use `query_params`

Reply:
For GET requests, data comes through in query params, and for POST/DELETE/PUT methods, data comes through request body, which is why we reference accordingly. Very rarely would you ever want to send a body with GET. Here's a nice SO link that goes over this topic: https://stackoverflow.com/questions/978061/http-get-with-request-body